sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1843,6 +1843,26 @@ std::vector<device_image_plain> ProgramManager::getSYCLDeviceImages(
   return DeviceImages;
 }
 
+void setSpecializationConstants(
+    const std::shared_ptr<device_image_impl> &InputImpl, RT::PiProgram Prog,
+    const plugin &Plugin) {
+  std::lock_guard<std::mutex> Lock{InputImpl->get_spec_const_data_lock()};
+  const std::map<std::string, std::vector<device_image_impl::SpecConstDescT>>
+      &SpecConstData = InputImpl->get_spec_const_data_ref();
+  SerializedObj SpecConsts = InputImpl->get_spec_const_blob_ref();
+
+  for (const auto &DescPair : SpecConstData) {
```

Comment:
nit: Can we use structured bindings for readability here?

Reply:
Absolutely!